### PR TITLE
[ISSUE #2882] Updated wallet / send transaction to latest design

### LINKED ISSUE #2881: Update wallet main screen
### User Story

As a user, I want to have access to latest wallet  UI changes 

### Description

Update wallet main screen so that main actions are displayed as a list. `transaction history` is now a first class action.
[Mockups](https://app.zeplin.io/project/5a55d587ba304eed4d2c942f/screen/5a589a0fccc89d66c86d6be8)

*Type*: Feature

### LINKED ISSUE #2882: Update wallet send transaction screen
User Story

As a user, I want to have access latest wallet UI changes

Description

Update wallet send transaction screen `recipient selection` and `amount` display. `advanced` screen and `token` selection will also be updated.

Type: Feature

### PR DESCRIPTION
**Pull request by jeluard**
fixes #2882

### Summary:

Upgrade `wallet` `send` transaction flow to reflect latest mockups.
When possible, UI `components` have been introduced and reused where relevant.
Some global style changes have been applied (like uniformed border radius).

### Steps to test:
- Open Status
- Go to wallet
- Make sure `send` transaction flows are fine, both from `wallet` and `chat`

status: ready

### CODE REVIEW

File: src/status_im/android/platform.cljs
Comment (flexsurfer) [inline src/status_im/android/platform.cljs]: can we move this into `ui/components/dialog` ?

File: src/status_im/ui/components/action_button/styles.cljs
Comment (flexsurfer) [inline src/status_im/ui/components/action_button/styles.cljs:24]: why not colors ns?

File: src/status_im/ui/components/list_selection.cljs
Comment (flexsurfer) [inline src/status_im/ui/components/list_selection.cljs]: maybe better to use dialog component? `:list-selection-fn` looks weird

**Review by flexsurfer — COMMENTED** [review]
wow, thats huge :)

File: src/status_im/chat/styles/message/message.cljs
Comment (goranjovic) [inline src/status_im/chat/styles/message/message.cljs:101]: In a smaller PR I'd ask you to rename :camelCase keys to :kebab-case 
Maybe that's another thing to do in a blanket refactor PR

**Review by goranjovic — APPROVED** [review]

File: src/status_im/ui/components/action_button/styles.cljs
Comment (jeluard) [reply src/status_im/ui/components/action_button/styles.cljs:24]: I didn't migrate all colors yet. It would make sense to have a task just for that.

File: src/status_im/chat/styles/message/message.cljs
Comment (jeluard) [reply src/status_im/chat/styles/message/message.cljs:101]: Right sounds like a good bounty!

Comment (jeluard) [reply src/status_im/chat/styles/message/message.cljs:101]: https://github.com/status-im/status-react/issues/3139

File: src/status_im/ui/components/action_button/styles.cljs
Comment (jeluard) [reply src/status_im/ui/components/action_button/styles.cljs:24]: #3140

File: src/status_im/ui/components/list_selection.cljs
Comment (jeluard) [reply src/status_im/ui/components/list_selection.cljs]: Sure! I didn't fully finished the refactoring for some reason .. ;)

**Comment by jeluard** [comment]
@flexsurfer I made changes according to your comments.

**Review by flexsurfer — APPROVED** [review]

**Comment by asemiankevich** [comment]
Branch: PR-3138
Android: https://i.diawi.com/Bzt1zY
iOS: https://i.diawi.com/3qGH6E

**Comment by asemiankevich** [comment]
iOS: `Send transaction`

1. Missing line under the `Send transaction` header
2. Increse footer space
image

iOS: `specify recipient`
1. exception shown
image

Android: `specify resipient`
1. Rename `Choose recipient` to `Recipient`
image
2. Cant scan code within Android. Tried to scan those ones:
http://prntscr.com/i5t35z
http://prntscr.com/i5t3ci (should throw an error saying its invalid wallet code)

`Advanced` screen for both platforms
image

Console picture is changed
image

Password hint is not hidden (Android)
image



<blockquote><div>Lightshot</div><div><strong><a href="https://prntscr.com/i5t35z">Screenshot</a></strong></div><div>Captured with Lightshot</div></blockquote>
<blockquote><div>Lightshot</div><div><strong><a href="https://prntscr.com/i5t3ci">Screenshot</a></strong></div><div>Captured with Lightshot</div></blockquote>

**Comment by jeluard** [comment]
Android: https://i.diawi.com/kzYGbW
iOS: https://i.diawi.com/gTjkrH

**Comment by Serhy** [comment]
Tested builds Android: https://i.diawi.com/kzYGbW and iOS: https://i.diawi.com/gTjkrH on Android 7.0 (Sansung J7 real device) and iPhoneX (with iOS 11.2.2)

Console icon is back, not hidden password hint - fixed, I can scan code within Android - cool!

We need to fix a couple of items though:
1) As in previous comment - on iOS only: tap on `Specify recipient...` ends up with an error:
error_no_item_label_blabla_ios

2) There is an error on **Android** when I choose recipient from the list of contacts: from Wallet main screen -> Send transaction -> tap Specify recipient -> tap 'Recent recipients' -> choose Jarrad -> `TypeError: null is not an object (evaluating 'e.lastIndexOf')`
elastindexof_20180129-144041

TF session with the error: https://app.testfairy.com/projects/4803622-status/builds/7592090/sessions/8/?accessToken=CDhKmObbVd9D8OrZOw0fE/654SQ

3) On iOS and Android we lost the ability to scroll to the bottom on 'Send transaction' screen to see the `Advanced ^` option while cursor in the `Specify amount` field. (but it's OK if `Advanced` option revealed showing 'Transaction Fee'). It's fine on develop. 
**To reproduce:** 
Main Wallet screen -> tap `Send transaction` -> tap in the `Specify amount...` editbox (so the cursor appears) -> scroll to bottom to find `Advanced` option to tap on it.
TF session: https://app.testfairy.com/projects/4803622-status/builds/7592090/sessions/10/?accessToken=1lnDYFhNgsbw-u1nyvP5gyhQG7A

The next are going for bounty:
- The alignment of values on Transaction Fee screen
- No Question mark on Transaction Fee screen
- No 'Reset to default' and 'Done' buttons 
- No line separating the header on iOS on 'Transaction Fee' and 'Send Transaction' screens

**Comment by goranjovic** [comment]
@Serhy

Branch: PR-3138
Android: https://i.diawi.com/pyYn8x
iOS: https://i.diawi.com/mUkvDV

^ fixes those three main bugs

**Comment by goranjovic** [comment]
@Serhy 

Rebased build - ready to test 

Branch: PR-3138
Android: https://i.diawi.com/BA3vnM
iOS: https://i.diawi.com/Las7dc

**Comment by Serhy** [comment]
@goranjovic thanks a lot! The Android: https://i.diawi.com/BA3vnM (27815295+) and iOS: https://i.diawi.com/Las7dc (0.0.2183) builds were tested on Android 7.0 (Samsung J7 real device), Android 6.0 (emulator) and iOS 11.2.2 (iPhoneX)

Above issues are fixed!  However, there are major issues appeared which we need to look at before merge:

1) On **Android and iOS** - there is a persistent white background screen when I choose the 'Scan QR code' option to specify recipient from 'Send transaction' screen instead of Camera view.
**To reproduce:**
 - Open Status
 - Navigate to Wallet -> Send transaction
 - Tap 'Specify recipient' -> 'Scan QR code' option
**Actual result:**
Android screen:
screenshot_20180201-194540

iOS screen:
image uploaded from ios 1

2) On **iOS only** the bottom tab bar (Home-Wallet-Profile) is not active on Wallet main screen. At the same time the Wallet button (from the Home-Wallet-Profile tab bar) - is "active" on 'Send transaction' screen. 

Check the steps to reproduce and TF session below:

- Open status
- Navigate to Wallet screen
- In the bottom tab bar: Tap Home button. Tap Profile button.
**Actual result:** nothing happens upon tap on any of Home/Profile buttons
- Tap on 'Send transaction'. (You navigated to 'Send transaction screen)
- Tap in the middle of the very bottom of screen (at the place where Wallet main would be if Home-Wallet-Profile main tab exist there)
**Actual result:** You navigated back to Wallet main screen
- Tap on 'Send transaction' 
- Tap 'Specify amount' and type valid amount
- Tap `Choose recipient' and choose Jarrad contact
**Actual result:** 'Sign later' and 'Sign transaction' buttons appeared. However nothing happens if to tap on them. If you tap on "imaginary" Wallet button in the middle of very bottom - you'll navigate to main Wallet screen.

TF session representing above steps:
https://app.testfairy.com/projects/4803590-status/builds/7617937/sessions/5/?accessToken=-MFDYcmZclK34Ze28wcThtrAYCw

Apparently, it may be consequences of the 3rd issue fix, that is the focus related to `Advanced ^` option described in https://github.com/status-im/status-react/pull/3138#issuecomment-361258386

**Review by oskarth — CHANGES_REQUESTED** [review]
@jeluard This PR is too big to review and merge. 72 files changes. Can we please break it up into smaller atomic commits so we decrease the risk of regressions and introduce smaller, more atomic changes into the code base?

**Comment by jeluard** [comment]
@oskarth I agree with the sentiment and in retrospect I should have split it (predates some discussion we had about this topic).

I'd rather go ahead with this one as it has been already been extensively reviewed  (@goranjovic worked on it some significant time) and tested. Splitting it would take significant time and probably introduce new regression / testing cycles.

**Comment by asemiankevich** [comment]
----------------
Branch: PR-3138
Android: https://i.diawi.com/QVny1o
iOS: https://i.diawi.com/qi4bpP

**Comment by goranjovic** [comment]
@jeluard @oskarth This PR already had 4 rebases with non-trivial conflicts. It took me a bloody while to merge that since both sides changed the logic of what the functions did. Effectively, had to write new code based on left hand and right hand examples. On the other hand, all the bad conflicts occurred in two namespaces alone - the central ones for the scope of this task. So the size expressed in number of changed files is really throwing dust in our eyes.

I think that putting this on hold any further (either monolithic or split) would only increase the window of opportunity for it to attract more conflicts. It should be either merged or rewritten from scratch.

**Comment by asemiankevich** [comment]
taskbar appears to be black, however white one is expected
image

2 issues above that @Serhy has mentioned are fixed

**Comment by asemiankevich** [comment]
Branch: PR-3138
Android: https://i.diawi.com/n3ufkR
iOS: https://i.diawi.com/fJSAcg

**Comment by asemiankevich** [comment]
@goranjovic iOS taskbar looks good. New issue found: can't tap `Done` button on iOS, keyboard is not hidden when pasting user address
image

**Comment by asemiankevich** [comment]
Suggest to merge it and have the following issues to be fixed as bounties:
1. Missing line on iOS under Send header
2. Missing `?` sign
3. Missing `~` sign
4. Keyboard is not hidden when pasting address (iOS)
5. All other UI issues that i will find during comaprison within mocks

issue with Missing `reset to defaults` and `done` buttons on transaction screen fee is also separate issue and will be fixed by core team (https://github.com/status-im/status-react/issues/3213)

@jeluard @goranjovic @Serhy thoughts?

**Comment by jeluard** [comment]
@asemiankevich Sounds good to me!

**Comment by jeluard** [comment]
@oskarth WDYT about accepting this one?

**Review by oskarth — APPROVED** [review]
Thanks everyone, will merge!

**Comment by oskarth** [comment]
@jeluard Can you please squash commits first?

**Comment by goranjovic** [comment]
@jeluard @oskarth Squashed

**Pull request merged**
